startup/32-data_export.py

```diff
@@ -0,0 +1,602 @@
+    df = pd.read_excel(fn, sheet_name=sheet_name, skiprows=skiprows, header=header, index_col=index_col)
+    input_dic = {}
+    for i in df.index:
+        try:
+            input_dic[i] = df.loc[i].dropna().tolist()
+        except AttributeError:
+            # temp = df.loc[i].to_dict(orient='split')
+            input_dic[i] = []
+            for j in range(df.loc[i].shape[0]):
+                ll = df.loc[i].iloc[j].dropna().tolist()
+                input_dic[i].append(ll)
+    return input_dic
+
+
+## Auto generate sample name with given prefix and infuse_rate
+## If prefix = None, 'Pre00', 'Pre01', 'Pre02', ... will be used.
+def _auto_name_sample(infuse_rates, prefix=None):
+    infuse_rates = np.asarray(infuse_rates)
+
+    if len(infuse_rates.shape) == 1:
+        infuse_rates = infuse_rates.reshape(1, infuse_rates.shape[0])
+
+    if prefix == None:
+        prefix_list = [f'Pre{i:02d}' for i in range(infuse_rates.shape[1])]
+    else:
+        prefix_list = prefix
+    
+    # prefix_array = np.asarray(prefix_list)
+
+    sample = []
+    for i in range(infuse_rates.shape[0]):
+        name = ''
+        # for j in range(infuse_rates.shape[1]):
+        for j in range(len(prefix_list)):
+            int_rate = int(round(float(infuse_rates[i][j]), 0))
+            name += f'{prefix_list[j]}_{int_rate:03d}_'
+        sample.append(name[:-1])
+    
+    return sample
+
+
+
+## Auto generate sample name with given prefix and directly reading pump rates
+## If prefix = None, 'Pre00', 'Pre01', 'Pre02', ... will be used.
+def _auto_name_sample2(pump_list, prefix=None):
+    
+    if type(pump_list[0]) == str:
+        pump_list = [eval(pump) for pump in pump_list]
+    else:
+        pass
+
+    get_rate = [pump.read_infuse_rate.get() for pump in pump_list]
+    get_unit = [pump.read_infuse_rate_unit.get() for pump in pump_list]
+    ruc_list = [rate_unit_converter(r0 = unit, r1 = 'ul/min') for unit in get_unit]
+    get_status = [pump.status.get() for pump in pump_list]
+
+    is_infusing = []
+    for pump_status in get_status:
+        if pump_status == 'Infusing':
+            is_infusing.append(1)
+        else:
+            is_infusing.append(0)
+
+    infuse_rates = np.asarray(get_rate) * np.asarray(ruc_list) * np.asarray(is_infusing)
+
+    if len(infuse_rates.shape) == 1:
+        infuse_rates = infuse_rates.reshape(1, infuse_rates.shape[0])
+
+    if prefix == None:
+        prefix_list = [f'Pre{i:02d}' for i in range(infuse_rates.shape[1])]
+    else:
+        prefix_list = prefix
+
+    sample = []
+    for i in range(infuse_rates.shape[0]):
+        name = ''
+        # for j in range(infuse_rates.shape[1]):
+        for j in range(len(prefix_list)):
+            int_rate = int(round(float(infuse_rates[i][j]), 0))
+            name += f'{prefix_list[j]}_{int_rate:03d}_'
+        sample.append(name[:-1])
+    
+    return sample
+
+
+
+def _exprot_rate_agent(metadata_dic, rate_label_dic, agent_data):
```

Comment:
```suggestion
def _export_rate_agent(metadata_dic, rate_label_dic, agent_data):
```

Check where it's used and fix the name there as well.